src/reducers/ui.js

```diff
@@ -24,7 +24,15 @@ const defaultState = new Immutable.Map().
   set('workspace', DEFAULT_WORKSPACE).
   set('notifications', new Immutable.Map()).
   set('topBar', new Immutable.Map({openMenu: null})).
-  set('saveIndicatorShown', false);
+  set('saveIndicatorShown', false).
+  set('assignmentSelector', new Immutable.Map({
+    openModal: false,
+    selectedCourse: '',
+    courses: new Immutable.List(),
+    dateInput: '',
```

Comment:
My thought in having both was since the date input field is a controlled component. Will look at your proposal re forms below.

Reply:
Yeah that makes total sense, I think it would also scan fine in the context of a clean separation between core assignment data and form properties as outlined above.